torchvision/prototype/models/depth/stereo/raft_stereo.py

```diff
@@ -0,0 +1,746 @@
+            layers=context_encoder_layers[:-1],
+            strides=context_encoder_strides,
+            norm_layer=nn.BatchNorm2d,
+        )
+        feature_base_encoder = base_encoder
+        context_base_encoder = base_encoder
+    else:
+        feature_base_encoder = BaseEncoder(
+            block=feature_encoder_block,
+            layers=feature_encoder_layers[:-1],
+            strides=feature_encoder_strides,
+            norm_layer=nn.InstanceNorm2d,
+        )
+        context_base_encoder = BaseEncoder(
+            block=context_encoder_block,
+            layers=context_encoder_layers[:-1],
+            strides=context_encoder_strides,
+            norm_layer=nn.BatchNorm2d,
+        )
+    feature_encoder = FeatureEncoder(
+        feature_base_encoder,
+        output_dim=feature_encoder_layers[-1],
+        shared_base=shared_encoder_weight,
+        block=feature_encoder_block,
+    )
+    context_encoder = MultiLevelContextEncoder(
+        context_base_encoder,
+        output_dim=context_encoder_layers[-1],
+        out_with_blocks=context_encoder_out_with_blocks,
+        block=context_encoder_block,
+    )
+
+    feature_downsampling_ratio = feature_encoder.base_downsampling_ratio
+
+    corr_block = CorrBlock1d(num_levels=corr_block_num_levels, radius=corr_block_radius)
+
+    motion_encoder = MotionEncoder(
+        in_channels_corr=corr_block.out_channels,
+        corr_layers=motion_encoder_corr_layers,
+        flow_layers=motion_encoder_flow_layers,
+        out_channels=motion_encoder_out_channels,
+    )
+    update_block = MultiLevelUpdateBlock(motion_encoder=motion_encoder, hidden_dims=update_block_hidden_dims)
+
+    # We use the largest scale hidden_dims of update_block to get the predicted depth
+    depth_head = FlowHead(
+        in_channels=update_block_hidden_dims[0],
+        hidden_size=flow_head_hidden_size,
+    )
+    if use_mask_predictor:
+        mask_predictor = MaskPredictor(
+            in_channels=update_block.hidden_dims[0],
+            hidden_size=mask_predictor_hidden_size,
+            out_channels=9 * feature_downsampling_ratio * feature_downsampling_ratio,
+        )
+    else:
+        mask_predictor = None
+
+    model = RaftStereo(
+        feature_encoder=feature_encoder,
+        context_encoder=context_encoder,
+        corr_block=corr_block,
+        update_block=update_block,
+        depth_head=depth_head,
+        mask_predictor=mask_predictor,
+        slow_fast=slow_fast,
+        num_iters=num_iters,
+        **kwargs,  # not really needed, all params should be consumed by now
+    )
+
+    if weights is not None:
+        model.load_state_dict(weights.get_state_dict(progress=progress))
+
+    return model
+
+
+class Raft_Stereo_Fast_Weights(WeightsEnum):
+    pass
+
+
+class Raft_Stereo_Weights(WeightsEnum):
+    pass
+
+
+def raft_stereo_fast(*, weights: Optional[Raft_Stereo_Fast_Weights] = None, progress=True, **kwargs) -> RaftStereo:
```

Comment:
Is this the `Slow-fast` variant? If yes, we should name as such because the "slow-fast" approach is something recognizable across papers.

Reply:
It use slow-fast gru, but not just that. I think the paper refer this model as "realtime model". It is specified on section 4.7 on RAFT-Stereo paper.

Reply:
Thank for the references. Page 8, just before section 4.7 reads:
> In table 6, the ”Slow-Fast” version of RAFT-Stereo updates the lowest, middle and highest resolution hidden states 30, 20 and 10 times, respectively, while ”Regular” updates each hidden state 32 times. 

If they refer as "slow-fast" to this variant, I recommend naming it slowfast. It's a recognizable name in the literature.

Reply:
I think the one they refer on the last part of section 4.6 is different model that they use for ablation study where they want to compare using `Slow-Fast` vs not using it (shown in table 6).
The model that I implement (and the one they provide pretrained weight on) has combination of techniques they explained on the ablation. Other than using slow-fast GRU, it use more downsampling, less GRU, shared backbone (kinda use all result of table 6).
In the original repo they call it `raftstereo-realtime.pth` (see last part of demos https://github.com/princeton-vl/RAFT-Stereo#demos ). 
Maybe I should call it `raft_stereo_realtime` instead?

Reply:
Sounds good to me.

Reply:
Sounds good, let's use `raft_stereo_realtime`